### PR TITLE
Deploy-To-ARM Demo

### PR DESCRIPTION
**Pull request by tshaiman**
- Adding README.md to explain about the deploy process
-  Supporting both "Button" deploy and Power shell deploy instruction
-  Adding a minimal Arm template for the demo.

### CODE REVIEW

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+
+# Overview
+
+In this Quick-Start you will create an Azure Analyzer for Media (a.k.a "AVAM") resource by using Arm Template (PREVIEW)
+The resource will be deployed to your subscription and will create the 'AVAM' resource based on parametrers presented on the avam.template file.
+
+> **Note:**
+> this sample is *not* for migrating an existing AVAM account to an ARM-Based AVAM account.
+> For a full documentation on AVAM API, visit the [AVAM Developer Portal](https://aka.ms/avam-dev-portal) page.
+> The current API Version is "2021-08-16-preview". Check this Repo from time to time to get updates on new API Versions.
```
Comment (tabenari) [inline ARM-Samples/Create-Video-Analyzer/README.md:10]: We already have 2021-09-01-preview up and running

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
+
+# Overview
+
+In this Quick-Start you will create an Azure Analyzer for Media (a.k.a "AVAM") resource by using Arm Template (PREVIEW)
+The resource will be deployed to your subscription and will create the 'AVAM' resource based on parametrers presented on the avam.template file.
```
[the remark this answers is not in the document]
Comment (tabenari) [reply ARM-Samples/Create-Video-Analyzer/README.md:5]: and maybe add the .json extension to the file name here

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+----
+
+### Option 2 : Deploy using Power Shell Script
+
+1. Open The [Template File](avam.template.json) file and inspect its content.
+2. Fill in the required parameters (see below)
+3. Run the Following Power Shell commands:
+
+* create new Resource group on the same location as your Azure Media Service account, using the [New-AzResourceGroup](https://docs.microsoft.com/en-us/powershell/module/az.resources/new-azresourcegroup) cmdlet.
```
Comment (tabenari) [inline ARM-Samples/Create-Video-Analyzer/README.md:32]: we recommend having the ams and the avam accounts in the same resource group I think, but as this is generic instructions, maybe leave it as is

File: ARM-Samples/Create-Video-Analyzer/README.md (hunk 1 above)
Comment (tshaiman) [reply ARM-Samples/Create-Video-Analyzer/README.md:10]: is there something changed ? This is why I've asked to see an Template sample.

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+### mediaServicesAccountId
+
+* type: string
+* description: The Resource Id of the Azure Media Service Account. 
+* required: true
+
+### managedIdentityId
+
+* type: string
+* description: The Application Id Of the Managed Identity used to grant access between AVAM resource and the Azure Media Service Account resource
```
Comment (tabenari) [inline ARM-Samples/Create-Video-Analyzer/README.md:70]: it is not the application id, but rather the user assigned managed identity uri (/subscriptions/...resourceGroups/...")

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+* required: true
+
+### location
+
+* type: string
+* description: Specifies the Azure location where the AVAM resource should be created.
+* required: false
+
+> **Note:**
+> You need to deploy Your Azure Video Analyzer For Media Account in the same region as the Azure Media Services account exists.
```
[the remark this answers is not in the document]
Comment (tshaiman) [reply ARM-Samples/Create-Video-Analyzer/README.md:59]: yep , I also hope @itay norman can get a technical writer

File: ARM-Samples/Create-Video-Analyzer/avam.template.json [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+                "description" : "The name of the AVAM resource"
+            },
+            "defaultValue": "avam-account"
+        },
+        "classicAccountId": {
+            "type": "string",
+            "metadata" : {
+                "description" : "When migrating from Classic Account to Arm accout this value indicates what is the classic Account reource Id."
+            },
+            "defaultValue" :""
```
Comment (tabenari) [inline ARM-Samples/Create-Video-Analyzer/avam.template.json:17]: defaultValue should be null if possible

File: ARM-Samples/Create-Video-Analyzer/avam.template.json [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+            "type": "string",
+            "metadata" : {
+                "description" : "The managed identity used to grant access to the Azure Media Service (AMS) account"
+            },
+            "defaultValue": "/subscriptions/<subscriptionId>/resourcegroups/<AVAMResourceGroup>/providers/Microsoft.ManagedIdentity/userAssignedIdentities/<MIName>"
+        },
+        "mediaServicesAccountId": {
+            "type": "string",
+            "metadata" : {
+                "description" : "The media Service Account Id. The Account needs to be created prior to the creation of this template"
```
Comment (tabenari) [inline ARM-Samples/Create-Video-Analyzer/avam.template.json:29]: I am not sure that account id is a good name, it is the account resource uri

File: ARM-Samples/Create-Video-Analyzer/avam.template.json [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+            "defaultValue": {
+                "team" : "backend",
+                "sku": "standard"
+            }
+        }
+    },
+    "resources": [
+        {
+            "type": "Microsoft.VideoIndexer/accounts",
+            "apiVersion": "2021-08-16-preview",
```
Comment (tabenari) [inline ARM-Samples/Create-Video-Analyzer/avam.template.json]: change to 2021-09-01-preview

File: ARM-Samples/Create-Video-Analyzer/avam.template.json (hunk 7 above)
Comment (tshaiman) [reply ARM-Samples/Create-Video-Analyzer/avam.template.json:29]: can you change it on the next Release ?

Comment (tabenari) [reply ARM-Samples/Create-Video-Analyzer/avam.template.json:29]: the name in the resource schema is "resourceId" (under mediaServices") which is fine. it is just the name of the parameter that is accountId that I am not sure about, and that you can change freely if you agree

File: ARM-Samples/Create-Video-Analyzer/README.md (hunk 1 above)
Comment (tabenari) [reply ARM-Samples/Create-Video-Analyzer/README.md]: yes things changed with the backend of ARM, RPaaS and AVAM between the versions. The template schema is the same, but the request must come in with the new version for it to work properly in the backend.

File: ARM-Samples/Create-Video-Analyzer/avam.template.json (hunk 7 above)
Comment (tshaiman) [reply ARM-Samples/Create-Video-Analyzer/avam.template.json:29]: done

**Review by tshaiman — COMMENTED** [review]
all comments fixed

**Review by tabenari — APPROVED** [review]

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
+
+# Overview
+
+In this Quick-Start you will create an Azure Analyzer for Media (a.k.a "AVAM") account by using Arm Template (PREVIEW)
+The resource will be deployed to your subscription and will create the 'AVAM' resource based on parametrers define in the avam.template file.
```
Comment (Itnorman) [inline ARM-Samples/Create-Video-Analyzer/README.md:5]:
```suggestion
The resource will be deployed to your subscription and will create the Video Analyzer for Media resource based on parameters defined in the avam.template file.

```

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
+
+# Overview
+
+In this Quick-Start you will create an Azure Analyzer for Media (a.k.a "AVAM") account by using Arm Template (PREVIEW)
+The resource will be deployed to your subscription and will create the 'AVAM' resource based on parametrers define in the avam.template file.
+
+> **Note:**
+> this sample is *not* for migrating an existing AVAM account to an ARM-Based AVAM account.
```
Comment (Itnorman) [inline ARM-Samples/Create-Video-Analyzer/README.md:8]:
```suggestion
> this sample is *not* for connecting an existing Video Analyzer for Media classic account to an ARM-Based Video Analyzer for Media account.

```

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
+
+# Overview
+
+In this Quick-Start you will create an Azure Analyzer for Media (a.k.a "AVAM") account by using Arm Template (PREVIEW)
+The resource will be deployed to your subscription and will create the 'AVAM' resource based on parametrers define in the avam.template file.
+
+> **Note:**
+> this sample is *not* for migrating an existing AVAM account to an ARM-Based AVAM account.
+> For a full documentation on AVAM API, visit the [AVAM Developer Portal](https://aka.ms/avam-dev-portal) page.
```
Comment (Itnorman) [inline ARM-Samples/Create-Video-Analyzer/README.md:9]:
```suggestion
> For full documentation on Video Analyzer for Media API, visit the [Developer Portal](https://aka.ms/avam-dev-portal) page.

```

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+```powershell
+New-AzResourceGroupDeployment -ResourceGroupName myResourceGroup -TemplateFile ./avam.template.json
+```
+
+## Parameters
+
+### name
+
+* type: string
+* description: Specifies the name of the new Azure Analyzer for Media Resource.
```
Comment (Itnorman) [inline ARM-Samples/Create-Video-Analyzer/README.md:49]:
```suggestion
* Description: Specifies the name of the new Video Analyzer for Media Resource.

```

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+
+```powershell
+New-AzResourceGroupDeployment -ResourceGroupName myResourceGroup -TemplateFile ./avam.template.json
+```
+
+## Parameters
+
+### name
+
+* type: string
```
Comment (Itnorman) [inline ARM-Samples/Create-Video-Analyzer/README.md:48]:
```suggestion
* Type: string

```

File: ARM-Samples/Create-Video-Analyzer/README.md (hunk 12 above)
Comment (Itnorman) [inline ARM-Samples/Create-Video-Analyzer/README.md:49]:
```suggestion
* Description: Specifies the name of the new Azure Video Analyzer for Media Resource.

```

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+* description: Specifies the name of the new Azure Analyzer for Media Resource.
+* required: true
+
+### location
+
+* type: string
+* description: Specifies the Azure location where the AVAM resource should be created.
```
Comment (Itnorman) [inline ARM-Samples/Create-Video-Analyzer/README.md:55]:
```suggestion
* Description: Specifies the Azure location where the AVAM resource should be created.

```

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+* required: false
```
Comment (Itnorman) [inline ARM-Samples/Create-Video-Analyzer/README.md:56]:
```suggestion
* Required: false

```

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+
+> **Note:**
+> You need to deploy Your Azure Video Analyzer For Media Account in the same region as where the Azure Media Services account exists.
+
+### mediaServiceAccountResourceId
+
+* type: string
```
Comment (Itnorman) [inline ARM-Samples/Create-Video-Analyzer/README.md:63]:
```suggestion
* Type: string

```

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+* description: The Resource Id of the Azure Media Service Account.
```
Comment (Itnorman) [inline ARM-Samples/Create-Video-Analyzer/README.md:64]:
```suggestion
* Description: The Resource Id of the Azure Media Service Account.

```

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 18]
```diff
@@ -0,0 +1,92 @@
...
+* required: true
+
+### managedIdentityId
+
+* type: string
+* description: The Resource Id Of the Managed Identity used to grant access between AVAM resource and the Azure Media Service Account resource
```
Comment (Itnorman) [inline ARM-Samples/Create-Video-Analyzer/README.md:70]:
```suggestion
* Description: The Resource Id Of the Managed Identity used to grant access between AVAM resource and the Azure Media Service Account resource

```

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 19]
```diff
@@ -0,0 +1,92 @@
...
+* required: true
+
+### tags
+
+* type: object
```
Comment (Itnorman) [inline ARM-Samples/Create-Video-Analyzer/README.md:75]:
```suggestion
* Type: object

```

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 20]
```diff
@@ -0,0 +1,92 @@
...
+* description: Array of Objects that represents custom user tags on the AVAM resource
```
Comment (Itnorman) [inline ARM-Samples/Create-Video-Analyzer/README.md:76]:
```suggestion
* Description: Array of Objects that represents custom user tags on the AVAM resource

```

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 21]
```diff
@@ -0,0 +1,92 @@
...
+
+----
+
+### Option 2 : Deploy using Power Shell Script
+
+1. Open The [Template File](avam.template.json) file and inspect its content.
+2. Fill in the required parameters (see below)
+3. Run the Following Power Shell commands:
+
+* create new Resource group on the same location as your Azure Media Service account, using the [New-AzResourceGroup](https://docs.microsoft.com/en-us/powershell/module/az.resources/new-azresourcegroup) cmdlet.
```
Comment (Itnorman) [inline ARM-Samples/Create-Video-Analyzer/README.md:32]:
```suggestion
* Create a new Resource group on the same location as your Azure Media Service account, using the [New-AzResourceGroup](https://docs.microsoft.com/en-us/powershell/module/az.resources/new-azresourcegroup) cmdlet.

```

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 22]
```diff
@@ -0,0 +1,92 @@
...
+
+* Deploy the template to the resoruce group using the [New-AzResourceGroupDeployment](https://docs.microsoft.com/en-us/powershell/module/az.resources/new-azresourcegroupdeployment) cmdlet.
+
+```powershell
+New-AzResourceGroupDeployment -ResourceGroupName myResourceGroup -TemplateFile ./avam.template.json
+```
+
+## Parameters
+
+### name
```
Comment (Itnorman) [inline ARM-Samples/Create-Video-Analyzer/README.md:46]:
```suggestion
### Name

```

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 23]
```diff
@@ -0,0 +1,92 @@
...
+> **Note:**
+> You need to deploy Your Azure Video Analyzer For Media Account in the same region as where the Azure Media Services account exists.
+
+### mediaServiceAccountResourceId
+
+* type: string
+* description: The Resource Id of the Azure Media Service Account.
+* required: true
+
+### managedIdentityId
```
Comment (Itnorman) [inline ARM-Samples/Create-Video-Analyzer/README.md:67]: Same here, if possible: Managed Identity ID

File: ARM-Samples/Create-Video-Analyzer/README.md [hunk 24]
```diff
@@ -0,0 +1,92 @@
...
+
+* type: object
+* description: Array of Objects that represents custom user tags on the AVAM resource
+ required: false
+
+### Notes
+
+## Reference Documentation
+
+If you're new to Video Analyzer, see :
```
Comment (Itnorman) [inline ARM-Samples/Create-Video-Analyzer/README.md:83]:
```suggestion
If you're new to Azure Video Analyzer for Media (formerly Video Indexer), see :

```

**Pull request closed without merging**